inc/lift/Request.h

```diff
@@ -69,13 +74,35 @@ class Request {
         http::Version http_version) -> void;
 
     /**
-     * Sets the timeout for this HTTP request.  This should be set before Perform() is called
+     * Sets the full timeout for this HTTP request.  This should be set before Perform() is called
      * or if this is an AsyncRequest before adding to an EventLoop.
+     * NOTE: If you have also set the response wait time, this is the should be longer than than the
+     * response wait time and should be used to keep connections alive. Otherwise, this is the only
+     * timeout that will be used
      * @param timeout The timeout for the request.
      * @return True if the timeout was set.
      */
-    auto SetTimeout(
-        std::chrono::milliseconds timeout) -> bool;
+    auto SetCurlTimeout(std::chrono::milliseconds timeout) -> bool;
+    
+    /**
+     * Get/set optional chrono milliseconds indicating how long the request has
+     * before it times out. This should only be set when you want to give the request
+     * a longer timeout to accommodate keep-alive connections that may have a slow
+     * response.
+     * @{
+     */
+    [[nodiscard]]
+    auto GetResponseWaitTime() const -> const std::optional<std::chrono::milliseconds>&
```

Comment:
So if curl timeout is 5 min and response-wait-time is 5 sec, could the request still take up to 5 min to timeout if connect or send is slow (or longer if it spends just under 5 min connecting, then up to 5 min on send)?

Reply:
Yes, that's right. The user will get their timeout in 5 seconds, and the curl timeout will still take 5 minutes (though it will timeout regardless of the state of the connection -- the curl timeout will not wait more than 5 minutes).